### ISSUE TITLE
ReadClient#empty? is broken

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#7568]
references #7476 #7524 #6779 

This bug manifest when the `queue.drain` is set to `true`. It seems its the only place where `ReadClient#empty?` is currently used. 

- [x] The `ReadClient#empty?` methods has an incorrect reference to the underlying queue to call `is_fully_acked?` which produces a runtime error as noted in #7476. The fix for this is trivial.

- [x] The `ReadClient#empty?` relies on the `is_fully_acked?` method which currently does not take into account the actual *empty* queue condition, as noted and discussed in #7524. For this I suggest we look for an acceptable fix to satisfy `ReadClient#empty?` without changing the semantic of `Queue.isFullyAcked()` which can *possibly* leads to other side effects as also discussed in #7524.  - PR in #7575 

- [x] Add `WrappedAckedQueue#empty?/` tests/specs. - PR in #7575

The `Queue.isFullyAcked()` semantic should be re-evaluated by taking into considerations potential side-effects and I suggest this is done into a separate issue #7570 where the issue raised concerning the related `HeadPage.hasSpace()` [call](https://github.com/elastic/logstash/blob/b3b9e605ac2e35eaedf0cc9fc0668ca6427fca19/logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java#L329) in #7524  is also looked at.

**Comment by colinsurprenant** [comment]
Note that the `queue.drain` was never documented so far so the risk for this affecting users is low.

**Comment by colinsurprenant** [comment]
PR #7575 basically fixes this issue. `ReadClient#empty?` now delegates to `WrappedAckedQueue#empty?` which in turn delegate to new `Queue.isEmpty()` method.

**Comment by colinsurprenant** [comment]
closing this will unblock #7583 to merge `queue.drain` documentation.

**Comment by colinsurprenant** [comment]
fixed by #7575
